Assets/GothicVR/Scripts/Creator/VobCreator.cs

```diff
@@ -40,34 +41,39 @@ private void Start()
             assetCache = SingletonBehaviour<AssetCache>.GetOrCreate();
         }
 
-        public void Create(GameObject root, WorldData world)
+        public async Task Create(GameObject root, WorldData world, Action<float> progressCallback)
         {
             if (!FeatureFlags.I.CreateVobs)
                 return;
-            
+
             var vobRootObj = new GameObject("Vobs");
             vobRootObj.SetParent(root);
             parentGos = new();
 
             CreateParentVobObject(vobRootObj);
-            CreateVobs(vobRootObj, world.vobs);
+            await CreateVobs(vobRootObj, world.vobs, progress => progressCallback?.Invoke(progress));
         }
 
         private void CreateParentVobObject(GameObject root)
         {
-            foreach (var type in (PxVobType[]) Enum.GetValues(typeof(PxVobType)))
+            foreach (var type in (PxVobType[])Enum.GetValues(typeof(PxVobType)))
             {
                 var newGo = new GameObject(type.ToString());
                 newGo.SetParent(root);
-                
+
                 parentGos.Add(type, newGo);
             }
         }
 
-        private void CreateVobs(GameObject root, PxVobData[] vobs)
+        private async Task CreateVobs(GameObject root, PxVobData[] vobs, Action<float> progressCallback, float progress = 0.5f)
         {
+            float totalProgress = progress;
```

Comment:
Why not putting it into a LoadingManager? I think of the following functions:
1. AddSegment(steps) --> LoadingSegment
2. LoadingSegment.addIncrement() // which will automatically update the loading bar UI.
(First idea. Please challenge it)

Cool thing about it: You can easily reuse it on MeshCreator etc.